ipc/provider/src/manager/evm/manager.rs

```diff
@@ -845,6 +846,36 @@ impl SubnetManager for EthSubnetManager {
         let receipt = pending_tx.retries(TRANSACTION_RECEIPT_RETRIES).await?;
         block_number_from_receipt(receipt)
     }
+
+    async fn get_top_down_nonce(
+        &self,
+        subnet: &SubnetID,
+        block_hash: &[u8],
+    ) -> Result<TopDownQueryPayload<u64>> {
+        let gateway_contract = gateway_getter_facet::GatewayGetterFacet::new(
+            self.ipc_contract_info.gateway_addr,
+            Arc::new(self.ipc_contract_info.provider.clone()),
+        );
+
+        let evm_subnet_id = gateway_getter_facet::SubnetID::try_from(subnet)?;
+        let (exists, nonce) = gateway_contract
+            .get_top_down_nonce(evm_subnet_id)
+            .block(BlockId::Hash(H256::from_slice(block_hash)))
+            .call()
+            .await?;
+
+        let nonce = if !exists {
+            log::warn!("subnet does not exists at block hash, return 0 nonce");
+            0
```

Comment:
Is there a particular reason why you opt to return 0 with a warning in the logs, instead of returning `None` or an `Err`? I looked at other cases in the file, and it looks in most other cases an error is returned: 
* [genesis_epoch](https://github.com/consensus-shipyard/ipc/blob/4110c41e575cb33bddeadf606049c5b2f5bf4f1a/ipc/provider/src/manager/evm/manager.rs#L99-L100) 
* [get_applied_topdown_nonce](https://github.com/consensus-shipyard/ipc/blob/4110c41e575cb33bddeadf606049c5b2f5bf4f1a/ipc/provider/src/manager/evm/manager.rs#L931-L932), 
* [prev_bottom_up_checkpoint_hash](https://github.com/consensus-shipyard/ipc/blob/4110c41e575cb33bddeadf606049c5b2f5bf4f1a/ipc/provider/src/manager/evm/manager.rs#L966)
* [get_block_hash](https://github.com/consensus-shipyard/ipc/blob/4110c41e575cb33bddeadf606049c5b2f5bf4f1a/ipc/provider/src/manager/evm/manager.rs#L170)

Because `get_block_hash` returns an `Err`, it means `get_validator_changeset` and `get_topdown_msgs` do so as well. You can argue that by the time you call this it will most certainly not return an error since it would have already failed on the previous two, but still, for consistency I think it should behave the same way.